src/net/ifaddrs/mod.rs

```diff
@@ -76,17 +76,43 @@ use self::iff_flags::IffFlags;
 mod sockaddr;
 use self::sockaddr::{IfAddrValue, sockaddr_to_ifaddrvalue};
 
-pub type InterfaceMap = HashMap<String, Vec<InterfaceAddr>>;
+mod tests;
+
+pub type InterfaceMap<'a> = HashMap<String, Vec<InterfaceAddr<'a>>>;
 
 /// Represents a handle into the operating system's knowledge about network
 /// interfaces present on the system. Allows the user to iterate over
 /// interface configurations.
-pub struct InterfaceAddrs {
+pub struct InterfaceAddrs<'a> {
     inner: *mut libc::ifaddrs,
-    current: Option<&'static libc::ifaddrs>,
+    current: Option<&'a libc::ifaddrs>,
+    do_free: bool,
 }
 
-impl InterfaceAddrs {
+impl<'a> InterfaceAddrs<'a> {
+    /// Creates an `InterfaceAddrs` from a raw pointer, without calling into
+    /// the `libc`.
+    ///
+    /// The destructor will not attempt to free memory on an InterfaceAddrs
+    /// created in this way.
+    ///
+    /// # Unsafety
+    /// The caller is responsible for making sure the given pointer is not
+    /// in invalid memory.
+    ///
+    /// # Errors
+    /// `Err(())` will be returned if `p` was void.
+    pub unsafe fn from_raw(p: *mut libc::ifaddrs) -> ::std::result::Result<Self, ()> {
+        match p.as_ref() {
+            Some(r) => Ok(Self {
```

Comment:
This fails to compile.  You should change `Self` to `InterfaceAddrs`

Reply:
It compiles for me? Something must be screwy with my Rustup, I'll change it. Thanks.

Reply:
You fixed the wrong line.  Just check the Travis build log to see the error I'm talking about.  If it works for you, that's probably because you're using a newer version of Rust than Travis.  We're using 1.13.0 in Travis.